studio/components/interfaces/Auth/AuthProvidersForm/ProviderForm.tsx

```diff
@@ -19,36 +20,40 @@ export interface ProviderFormProps {
 
 const ProviderForm = ({ provider }: ProviderFormProps) => {
   const [open, setOpen] = useState(false)
-  const { authConfig, ui } = useStore()
-  const { project: selectedProject } = useProjectContext()
-  const { ref } = useParams()
+  const { ui } = useStore()
+  const { ref: projectRef } = useParams()
+  const { data: config } = useAuthConfigQuery({ projectRef })
+  const { mutate: updateAuthConfig } = useAuthConfigUpdateMutation()
+
   const doubleNegativeKeys = ['MAILER_AUTOCONFIRM', 'SMS_AUTOCONFIRM']
   const canUpdateConfig = useCheckPermissions(PermissionAction.UPDATE, 'custom_config_gotrue')
 
-  const { data: customDomainData } = useCustomDomainsQuery({ projectRef: ref })
+  const { data: customDomainData } = useCustomDomainsQuery({ projectRef })
 
   const generateInitialValues = () => {
     const initialValues: { [x: string]: string } = {}
+    // the config is already loaded through the parent component
+    if (!config) {
```

Comment:
hmm im not sure i follow - what do you mean by config is already loaded through the parent component? through the `provider` props?

am also thinking the short circuit here should return initialValues as an empty object too

Reply:
I've refactored this component to not deal with loading state, but access the `authConfig` from it's parent component.